samples/compiler_plugins/simple_io_sample/src/simple_io_sample/Transforms/LegalizeSimpleIO.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "simple_io_sample/IR/SimpleIOOps.h"
+#include "simple_io_sample/Transforms/Passes.h"
+
+#define GEN_PASS_DEF_LEGALIZESIMPLEIO
+#include "simple_io_sample/Transforms/Passes.h.inc"
+
+namespace mlir::iree_compiler::IREE::SimpleIO {
+namespace {
+
+class LegalizeSimpleIOPass
+    : public ::impl::LegalizeSimpleIOBase<LegalizeSimpleIOPass> {
+ public:
+  void runOnOperation() override {
+    auto *context = &getContext();
+    // Add imports.
+    auto m = getOperation();
+    auto importBuilder = OpBuilder::atBlockBegin(m.getBody());
```

Comment:
You may need same trick as LLVM translate side where one checks first if it exists already.

Reply:
Yeah, this is not very robust. I'm going to extend the sample to something interesting in the next step and will blow this away.